src/ECModel.jl

```diff
@@ -506,27 +506,27 @@ function data_sankey(ECModel::AbstractEC;
     production_us = calculate_production(ECModel)
 
     # definition of the ids of the resources
-    market_id_from = 1
-    market_id_to = length(user_set) + 3
+    user_id_from(x) = x
+    market_id_sell = length(user_set) + 1
+    market_id_buy = length(user_set) + 3
     community_id = length(user_set) + 2
-    user_id_from(x) = x + 1
-    user_id_to(x) = x + length(user_set) + 3
+    user_id_to(x) = x + length(user_set) + 4
 
     # specify the desired location of each entity id in the layers of the Sankey diagram
     # On the first layers all users that produce energy and the market when buying
     # On the second layer "Community"
     # On the third layer all users that consume energy and the market when selling
     node_layer = Dict([
-        market_id_from => 1;
         [user_id_from(id) => 1 for id in 1:length(user_set)];
-        community_id => 2;
-        market_id_to => 3;
-        [user_id_to(id) => 3 for id in 1:length(user_set)];
+        market_id_sell => 2;
+        community_id => 3;
+        market_id_buy => 4;
+        [user_id_to(id) => 5 for id in 1:length(user_set)];
     ])
 
     # create an ordering of the labels
-    order_list = Dict(id => id +1 for id in 1:length(name_units) if market_id_to != id)
-    push!(order_list, length(name_units)=>market_id_to)  # move market to before the users
+    order_list = Dict(id => id +1 for id in 1:length(name_units) if market_id_sell != id)

```

Comment:
Most likely you may need to delete the " if market_id_sell != id" and the following line is no more needed